engine/src/main/java/org/hibernate/search/bridge/util/impl/NumericFieldUtils.java

```diff
@@ -118,9 +118,12 @@ public static boolean requiresNumericRangeQuery(Object value) {
 			return false;
 		}
 		return value instanceof Double ||
+				value instanceof Byte ||
+				value instanceof Short ||
 				value instanceof Long ||
 				value instanceof Integer ||
 				value instanceof Float ||
+				value instanceof Date ||
```

Comment:
Ok, that sounds a bit scary :) Yes, adding some comment would be good. The method appears to be very general-purpose, so it was unexpected it's only used with ISPN.